### ISSUE TITLE
[Feature]: Separate label for Canada Country Code on Phone Input Fields

### ISSUE THREAD

**Issue by jdglacier** [appsmithorg/appsmith#18136]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

On the Phone Input Widget, add a separate country code to the drop-down list  for Canada to ensure the label is properly displayed on forms. 

![image](https://user-images.githubusercontent.com/111010738/200621069-bdb32e1c-be8c-41d7-8f6b-1dff8f06f127.png)

### Why should this be worked on?

Even though Canada and the United States share the same country code, the label is displayed as 'US'. Adding a separate option for Canadian phone numbers would help avoid confusion for Canadian users of Appsmith.

**Comment by somangshu** [comment]
@vsvamsi1 have we figured what is going to be our approach here?

**Comment by vsvamsi1** [comment]
Yeah i have, will be using the countryCode instead of the ISD code. Canada and USA share the same ISD code, so its difficult to distinguish between both. Hence using the countryCode.

**Comment by dilippitchika** [comment]
Hey i didn't understand the approach, can you explain what the difference is between country and isd code?

**Comment by dilippitchika** [comment]
Discussed with @vsvamsi1 
- We need to create a new switch to let people select if they want to give default as country code or dial code
- People who want to show canada can use the country code ("CA") instead of dial code (+1)
- Migrate the current implementation to dial code format

This looks like a bigger effort, we will come back to this later. CC @somangshu

**Comment by mickaelboudreau** [comment]
Hello,

As a Canadian organization, we would like to have a distinct Canadian flag for the +1 country code in the phone number input. 
Currently, only the US flag is shown, which can be confusing for our users. 

Adding a Canadian flag will improve clarity and user experience for Canadian users.
